File: TyreVibes/Models/AddressSuggestion.swift
import Foundation

struct AddressResponse: Codable {
    let status: String
    let resp: [AddressSuggestion]
    let time: Int
}

struct AddressSuggestion: Codable, Identifiable {
    let iso3: String
    let level: String
    let id: Int
    let score: Double
    let country: String
    let region: String
    let province: String
    let city: String
    let district1: String?
    let zipcode: String
    let street: String
    let chk: String
}

File: android/app/src/main/java/it/tyrevibes/app/core/model/AddressSuggestion.kt
package it.tyrevibes.app.core.model

import kotlinx.serialization.Serializable

@Serializable
data class AddressResponse(
    val status: String,
    val resp: List<AddressSuggestion>,
    val time: Int
)

@Serializable
data class AddressSuggestion(
    val iso3: String,
    val level: String,
    val id: Int,
    val score: Double,
    val country: String,
    val region: String,
    val province: String,
    val city: String,
    val district1: String? = null,
    val zipcode: String,
    val street: String,
    val chk: String
)
